ethereum/contracts/Scale.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+contract Scale {
+
+    // Read a byte at a specific index and return it as type uint8
+    function readByteAtIndex(bytes memory data, uint8 index)
+        internal
+        pure
+        returns (uint8)
+    {
+        return uint8(data[index]);
+    }
+
+    // Reverse uint8[] array in place
+    function reverse(uint8[] memory arr)
+        internal
+        pure
+        returns(uint8[] memory)
+    {
+        for (uint i = 0; i < arr.length/2; i++) {
+            uint8 current = arr[i];
+            uint256 otherIndex = arr.length - i - 1;
+            arr[i] = arr[otherIndex];
+            arr[otherIndex] = current;
+        }
+        return arr;
+    }
+
+    // Convert an uint8[] array to uint256
+    function uint8ArrayToUint256(uint8[] memory arr)
+        internal
+        pure
+        returns (uint256)
+    {
+        uint256 number;
+        for(uint i = 0; i < arr.length; i++){
+            number = number+uint(arr[i])*(2**(8*(arr.length-(i+1))));
+        }
+        return number;
+    }
+
+      // Convert bytes (big endian) to little endian format
+    function bytesToLittleEndian(bytes memory arr)
+        internal
+        pure
+        returns (uint256)
+    {
+        uint256 number;
+        for(uint i = 0; i < arr.length; i++){
+            number = number+uint(uint8(arr[i]))*(2**(8*(arr.length-(i+1))));
```

Comment:
Could we have spaces between binary operators to make the code less dense and improve readability?

Reply:
Fixed in https://github.com/Snowfork/polkadot-ethereum/commit/d3d5a2bc7a79b6949357ec8082aae20ce6b9c1b6